src/commands/cmd_config.c

```diff
@@ -80,34 +82,63 @@ void _Config_set(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
 	// field is not allowed to be reconfigured
 	if(!configurable_field) {
 		RedisModule_ReplyWithError(ctx, "Field can not be re-configured");
-		return;
+		return true;
 	}
 
 	// set the value of given config
-	RedisModuleString *value = argv[3];
-	const char *val_str = RedisModule_StringPtrLen(value, NULL);
+	const char *val_str = RedisModule_StringPtrLen(val, NULL);
 
 	if(Config_Option_set(config_field, val_str)) {
 		RedisModule_ReplyWithSimpleString(ctx, "OK");
 	} else {
 		RedisModule_ReplyWithError(ctx, "Failed to set config value");
+		return true;
 	}
+
+	return false;
+}
+
+int _Config_array_set(RedisModuleCtx *ctx, RedisModuleString **argv, int argc, bool dryrun) {
+	// To ensure atomicity we check if configuration can be setted.
+	if(dryrun) {
+		Config_Clone();                 // Clone the config for the case of error.
+		Config_Unsubscribe_Changes();   // Unsubscribe from configuration changes to avoid propagating the changes.
+	}
+
+	// set configuration for each requested one
+	for(int i = 2; i < argc; i += 2) {
+		RedisModuleString *key = argv[i];
+		RedisModuleString *val = argv[i+1];
+		if(_Config_set(ctx, key, val)) {
+			// On error quit the operation.
+			ASSERT(dryrun);
+			return -1;
+		}
+	}
+
+	if(dryrun) Config_RestoreFromClone();
+	return 0;
 }
```

Comment:
Sorry, my intension was to introduce `Config_DryRun` to `config.h`
`_Config_set` should call `Config_DryRun` on every input and only commit changes when all calls to `Config_DryRun` pass.